eos-usb-reflash/factory-test/start.sh

```diff
@@ -6,7 +6,29 @@
 set -e
 mydir=$(dirname $(readlink -f $0))
 
-KERNEL_VERSION="$(uname -r)"
+# If this script is running from an exFAT filesystem we should try to manually
+# load the kernel module, as on all systems prior to EOS 4 the backing device
+# will have been mounted using the FUSE exFAT driver (which is not available in
+# the initramfs we are about to pivot to).
+
+# findmnt returns fuseblk for FSTYPE if the FUSE driver is in use, so we need
+# to find the device node and ask lsblk.
+BACKING_DEV=$(findmnt --noheadings --output SOURCE --target $mydir)
+BACKING_FSTYPE=$(lsblk --noheadings --output FSTYPE $BACKING_DEV)
+
+if [ $BACKING_FSTYPE == "exfat" ] ; then
+    echo "Running from an exFAT filesystem, trying to load the kernel module..."
+    set +e
+    modprobe exfat
+    exfat_loaded=$?
+    if [ $exfat_loaded -ne 0 ] ; then
+        echo -n "This version of EOS does not have in-kernel exFAT support. "
+        echo -n "You need a device formatted with a compatible filesystem "
+        echo "(ext2 / ext4) to reflash this computer."
```

Comment:
Is regular old fat32 supported? That's an in-kernel module that's been around forever.

Reply:
FAT32 should technically work (I haven't tested it), but it has a file size limitation of 4GB, so in practice we won't be able to use it with any image except for the `base` personality (already 2.6GB) atm. Assuming it works, do you think we should also mention it on the comment?
```suggestion
        echo "(ext2 / ext4, or FAT32 if your image file has less than 4GB) to reflash this computer."
```

Reply:
Great point… probably not worth mentioning then. I would expect that users of this tool are more likely to be using more substantial images.